app/controllers/publish/courses/study_sites_controller.rb

```diff
@@ -55,6 +57,12 @@ def current_step
       def error_keys
         [:study_sites]
       end
+
+      def study_site_params
+        return { study_site_ids: nil } if params[:publish_course_study_site_form][:study_site_ids].all?(&:empty?)
```

Comment:
Why are we returning `nil` here? `present?` is used to determine if the value is valid or not later and that returns false for `nil` and empty array.

Reply:
I think we need to return nil here now, as I’ve followed your suggestion to use `validates :study_site_ids, presence: true` in the form.

Reply:
I looked into this. It appears Rails returns `[""]` for `study_site_ids` when no selection is made. This is probably an issue with http forms where the "key" must have some value in order to be sent in the request.

We could `params.require(:publish_course_study_site_form).permit(study_site_ids: []).compact_blank`.

```ruby
[1] pry(main)> h = { one: 1, nill: nil, empt: "" }
=> {:one=>1, :nill=>nil, :empt=>""}
[2] pry(main)> params = ActionController::Parameters.new(h)
=> #<ActionController::Parameters {"one"=>1, "nill"=>nil, "empt"=>""} permitted: false>
[3] pry(main)> params.compact_blank
=> #<ActionController::Parameters {"one"=>1} permitted: false>
[4] pry(main)> params
=> #<ActionController::Parameters {"one"=>1, "nill"=>nil, "empt"=>""} permitted: false>
[5] pry(main)> params.permit("one", "nill", "empt")
=> #<ActionController::Parameters {"one"=>1, "nill"=>nil, "empt"=>""} permitted: true>
[6] pry(main)> params.permit("one", "nill", "empt").compact_blank
=> #<ActionController::Parameters {"one"=>1} permitted: true>
```

Reply:
I tried this, it doesn't return `nil`, so the form doesn't validate when submitting empty. 

Also keeping it as it is helps keep things consistent with what we already have in the codebase, e.g: 

- https://github.com/DFE-Digital/publish-teacher-training/blob/0686327a0d76956c0d157b36a8021987544a7452/app/controllers/publish/courses/schools_controller.rb#L70

- https://github.com/DFE-Digital/publish-teacher-training/blob/0686327a0d76956c0d157b36a8021987544a7452/app/controllers/publish/courses/study_mode_controller.rb#L34
 
- https://github.com/DFE-Digital/publish-teacher-training/blob/0686327a0d76956c0d157b36a8021987544a7452/app/controllers/publish/providers/skilled_worker_visa_controller.rb#L32
 
- https://github.com/DFE-Digital/publish-teacher-training/blob/0686327a0d76956c0d157b36a8021987544a7452/app/controllers/publish/providers/student_visa_controller.rb#L32